source/extensions/access_loggers/grpc/grpc_access_log_utils.cc

```diff
@@ -300,16 +301,11 @@ void Utility::extractCommonAccessLogProperties(
   }
 
   for (const auto& key : config.filter_state_objects_to_log()) {
-    if (auto state = stream_info.filterState().getDataReadOnlyGeneric(key); state != nullptr) {
-      ProtobufTypes::MessagePtr serialized_proto = state->serializeAsProto();
-      if (serialized_proto != nullptr) {
-        auto& filter_state_objects = *common_access_log.mutable_filter_state_objects();
-        ProtobufWkt::Any& any = filter_state_objects[key];
-        if (dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()) != nullptr) {
-          any.Swap(dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()));
-        } else {
-          any.PackFrom(*serialized_proto);
-        }
+    if (!(extractFilterStateData(stream_info.filterState(), key, common_access_log))) {
+      if (stream_info.upstreamInfo().has_value() &&
+          stream_info.upstreamInfo()->upstreamFilterState().get() != nullptr) {
+        extractFilterStateData(*(stream_info.upstreamInfo()->upstreamFilterState().get()), key,
+                               common_access_log);
```

Comment:
nit: Seems our clang-tidy still not work as expected, or it should complain the redundant `get()` here.

`stream_info.upstreamInfo()->upstreamFilterState() != nullptr`
`*stream_info.upstreamInfo()->upstreamFilterState()`